src/collection-facets/more-facets-content.ts

```diff
@@ -45,59 +48,91 @@ import {
 } from '../utils/analytics-events';
 import './toggle-switch';
 import { srOnlyStyle } from '../styles/sr-only';
+import {
+  mergeSelectedFacets,
+  sortBucketsBySelectionState,
+  updateSelectedFacetBucket,
+} from '../utils/facet-utils';
 
 @customElement('more-facets-content')
 export class MoreFacetsContent extends LitElement {
   @property({ type: String }) facetKey?: FacetOption;
 
-  @property({ type: String }) facetAggregationKey?: FacetOption;
-
   @property({ type: String }) query?: string;
 
   @property({ type: Object }) filterMap?: FilterMap;
 
-  @property({ type: Object }) modalManager?: ModalManagerInterface;
-
-  @property({ type: Object }) searchService?: SearchServiceInterface;
-
   @property({ type: String }) searchType?: SearchType;
 
   @property({ type: Object }) pageSpecifierParams?: PageSpecifierParams;
 
   @property({ type: Object })
   collectionTitles?: CollectionTitles;
 
+  /**
+   * Maximum number of facets to show per page within the modal.
+   */
+  @property({ type: Number }) facetsPerPage = 35;
+
+  /**
+   * Whether we are waiting for facet data to load.
+   * We begin with this set to true so that we show an initial loading indicator.
+   */
+  @property({ type: Boolean }) facetsLoading = true;
+
+  /**
+   * The set of pre-existing facet selections (including both selected & negated facets).
+   */
   @property({ type: Object }) selectedFacets?: SelectedFacets;
 
   @property({ type: String }) sortedBy: AggregationSortType =
     AggregationSortType.COUNT;
 
-  @property({ type: Object, attribute: false })
-  analyticsHandler?: AnalyticsManagerInterface;
-
-  @state() aggregations?: Record<string, Aggregation>;
+  @property({ type: Object }) modalManager?: ModalManagerInterface;
 
-  @state() facetGroup?: FacetGroup[] = [];
+  @property({ type: Object }) searchService?: SearchServiceInterface;
 
-  @state() facetGroupTitle?: String = '';
```

Comment:
This property was also redundant -- the `facetGroup` already contains its title as a property.